src/controllers/userController.js

```diff
@@ -30,7 +42,37 @@ const updateUser = async(req,res) => {
     res.status(StatusCodes.OK).json({ user: tokenUser });
 };
 
+const updateUserPassword = async(req,res) => { // Function to update the current user's password
+    const { oldPassword, newPassword } = req.body;
+    if (!oldPassword || !newPassword) { // Check if oldPassword and newPassword are provided
+        throw new CustomError.BadRequestError('Please provide both values');
+    }
+    const user = await User.findOne({ _id: req.user.userId }); // Find the user in the database based on the current user's ID
+    // Compare the old password provided with the user's current password
+    const isPasswordCorrect = await user.comparePassword(oldPassword);
+    if(!isPasswordCorrect) { // Throw an UnauthenticatedError if the old password is incorrect
+        throw new CustomError.UnauthenticatedError('Invalid Credentials');
+    }
+    user.password = newPassword; // Update the user's password
```

Comment:
Is this saving the user's new password in plain text? I think it's not hashing it since I don't see `argon` being used here or in the User model.

Reply:
57-58 - Hash the new password